esp-hal/src/uart.rs

```diff
@@ -1859,12 +1801,86 @@ mod asynch {
 
         /// Create a new UART instance with defaults in [`Async`] mode.
         pub fn new_async(uart: impl Peripheral<P = T> + 'd, clocks: &Clocks) -> Self {
-            Self::new_async_with_config(
-                uart,
-                Default::default(),
-                None::<TxRxPins<'_, NoPinType, NoPinType>>,
-                clocks,
-            )
+            Self::new_async_with_config(uart, Default::default(), clocks)
+        }
+
+        /// Configure TX and RX pins
+        pub fn with_tx_rx<TX: OutputPin, RX: InputPin>(
```

Comment:
We can remove `with_tx_rx` in favour of always taking two pins in the Uart constructor